external/mqtt/src/test/scala/org/apache/spark/streaming/mqtt/MQTTStreamSuite.scala

```diff
@@ -121,8 +130,18 @@ class MQTTStreamSuite extends FunSuite with Eventually with BeforeAndAfter {
         val message: MqttMessage = new MqttMessage(data.getBytes("utf-8"))
         message.setQos(1)
         message.setRetained(true)
+
+        def tryPublish(times: Int): Unit =
```

Comment:
Defining this function inside the "if" looks a little weird, too nested. Might be better to just have it in for loop. Code would look less complicated.